nntrainer/models/neuralnet.cpp

```diff
@@ -560,6 +464,34 @@ sharedConstTensors NeuralNetwork::inference(sharedConstTensors X) {
   return out;
 }
 
+int NeuralNetwork::assignMem() {
```

Comment:
I think the strategy is good though, IMO, `assignMem` should be private and used only internally because this is not part of the API. Meaning that if someone is creating a new type of model, they shouldn't care about this method.

Reply:
yes. but like always, unittest is matter. on second thought, they might have to care about this method to set memory differently.

Reply:
ok